### ISSUE TITLE
There is a clear difference between downloading files using binary files and downloading files directly using yt-dlp in scripts

### ISSUE THREAD

**Issue by es3344520** [yt-dlp/yt-dlp#10473]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

On the YouTube platform, the same video files, the same format, are downloaded using mac's yt-dlp binary files and python scripts are downloaded using the yt-dlp library. The video file downloaded by the binary command is smaller than that downloaded by the library, and the file size may differ by 5M-10M.   
Although the number of video minutes is not less, the video file sizes are different. The yt-dlp parameters used are basically the same. 

### Provide verbose output that clearly demonstrates the problem

- [ ] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [ ] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
On the YouTube platform, the same video files, the same format, are downloaded using mac's yt-dlp binary files and python scripts are downloaded using the yt-dlp library. The video file downloaded by the binary command is smaller than that downloaded by the library, and the file size may differ by 5M-10M.   
Although the number of video minutes is not less, the video file sizes are different. The yt-dlp parameters used are basically the same.
```

**Comment by es3344520** [comment]
Moreover, I also found that when downloading binary files via python scripts, the download speed is much slower than when using the library directly

**Comment by seproDev** [comment]
Could you please provide verbose logs for both the CLI usage and library usage?

Using the same version of yt-dlp, with the same arguments, the output of the library and the CLI should be identical.

**Comment by es3344520** [comment]
Currently, the two scripts call yt-dlp in different ways. One directly calls the yt-dlp binary file, and the other directly references the yt-dlp repository in the script.  The other parameters and code are basically the same. 
Moreover, it is the same link downloaded, and the last two files are of different sizes.

**Comment by seproDev** [comment]
If you can't provide a verbose log for both, I can not help you. You are likely downloading different formats, but there is no way for me to tell without seeing the logs.

**Comment by es3344520** [comment]
> If you can't provide a verbose log for both, I can not help you. You are likely downloading different formats, but there is no way for me to tell without seeing the logs.

I have found the problem. For the same script code, although the method of calling yt-dlp is different, the audio sampling rate is different when downloading the same link. The script that calls yt-dlp binary downloads 44khz, while the script that calls the yt-dlp library downloads 48khz.   But what was weird was that when I changed another link, the audio sampling rates downloaded by the two scripts were the same, and I was suddenly confused.
 
But there is still one difficult problem at present, that is, the download response of the script that calls the yt-dlp binary file is much slower than that of calling the yt-dlp library.
